### PR TITLE
Move Pinot server APIs to jax-rs

### PR DESCRIPTION
**Pull request by antumbde**
Use jersey based jax-rs style APIs for Pinot servers.
This change is to provide more modern and convenient
way to create REST APIs. We continue to integrate with
Swagger.

### CODE REVIEW

File: pinot-server/src/main/resources/api/index.html [hunk 1]
```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
```
Comment (jfim) [inline pinot-server/src/main/resources/api/index.html]: This looks like the stock swagger index file, if so, you should pull it from the webjar or document the changes from the stock version.

Comment (antumbde) [inline pinot-server/src/main/resources/api/index.html]: added comment

File: pinot-server/src/main/java/com/linkedin/pinot/server/api/resources/MmapDebugResource.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * Debug endpoint to check memory allocation.
+ */
+@Api(value="debug", description="Debug information", tags = "Debug")
+@Path("debug")
+public class MmapDebugResource {
+
+  public static class AllocationInfo {
+    String context;
+    String type;
+    int size;
```
Comment (mcvsubbu) [inline pinot-server/src/main/java/com/linkedin/pinot/server/api/resources/MmapDebugResource.java]: I suppose this shows up as 'size' in the output? Would it be better to use sizeInBytes (or as appropriate)?

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/AdminApiApplication.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  }
+
+  public boolean start(int httpPort) {
+    if (httpPort <= 0 ) {
+      LOGGER.warn("Invalid admin API port: {}. Not starting admin service", httpPort);
+      return false;
+    }
+
+    baseUri = URI.create("http://0.0.0.0:" + Integer.toString(httpPort) + "/");
```
Comment (mcvsubbu) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/AdminApiApplication.java]: Does this work with ipv6?

Comment (antumbde) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/AdminApiApplication.java]: this won't work with ipv6. I'll file separate ticket to make this interface configurable.

File: pinot-server/src/main/java/com/linkedin/pinot/server/api/resources/MmapDebugResource.java (hunk 2 above)
Comment (antumbde) [inline pinot-server/src/main/java/com/linkedin/pinot/server/api/resources/MmapDebugResource.java]: Retaining compatibility with existing response format.

**Pull request merged**